programs/client/Client.cpp

```diff
@@ -706,6 +706,43 @@ bool Client::processWithFuzzing(const String & full_query)
         return true;
     }
 
+    const auto & settings = global_context->getSettingsRef();
+    const Dialect & dialect = settings.dialect;
+    String old_dialect;
+    switch (dialect)
+    {
+        case DB::Dialect::kusto:
+            old_dialect = "kusto";
+            break;
+        case DB::Dialect::clickhouse:
+            old_dialect = "clickhouse";
+            break;
+        case DB::Dialect::prql:
+            old_dialect = "prql";
+            break;
+    }
+
+    if (auto *q = orig_ast->as<ASTSetQuery>())
+    {
+        auto *setDialect = q->changes.tryGet("dialect");
+        if (setDialect)
+        {
+            old_dialect =  setDialect->get<String>();
+        }
+    }
+
+    //setting dialect to clickhouse during query fuzzing, restore dialect to original value after fuzzing
+
+    SCOPE_EXIT_SAFE({
+            global_context->setSetting("dialect", old_dialect);
+    });
+
+    if (dialect != DB::Dialect::clickhouse)
+    {
+        SettingChange new_setting("dialect", "clickhouse");
+        global_context->applySettingChange(new_setting);
+    }
+
```

Comment:
@kashwy I wonder why do we need it at all - this fuzzing functionality is introduced for testing - do you have problems without this addition? it seems somewhat off and I would prefer to remove it if it's not absolutely necessary

Reply:
we got fuzzy test failure without this. the reason behind is that, fuzzing tests are generated from AST which are SQLs. while during KQL test , the dialect has been set to 'kusto' , so the fuzzing SQL wont work

Reply:
I guess we can avoid this issue just returning true if it's a kusto dialect - the same as above at 706